testsuite/tests/match-side-effects/check_partial.ml

```diff
@@ -0,0 +1,113 @@
+(* TEST
+ flags = "-dlambda";
+ expect;
+*)
+
+type _ t =
+  | Int : int -> int t
+  | True : bool t
+  | False : bool t
+
+let lazy_total : _ * bool t -> int = function
+  | ({ contents = _ }, True) -> 0
+  | ({ contents = lazy () }, False) -> 12
+(* This pattern-matching is in fact total: a Match_failure case is
+   not necessary for soundness. *)
```

Comment:
```suggestion
(* This pattern-matching is in fact total: the Match_failure case
   conservatively inserted by the pattern match compiler is not
   necessary for soundness. *)
```

To distinguish this case more clearly from the "This pattern-matching is partial: a Match_failure case is necessary for soundness." comment below on a test where
the case is *not* inserted.

Reply:
I had more explanations before but I repeat those sentences for several examples and the duplication was bad. To take your suggestion into account I added an explanatory paragraph at the beginning of the test file.